### DISCUSSION TITLE
Can we retain layout when external monitors are plugged in/out?

### DISCUSSION THREAD

**Discussion by debaraj-barua** [rxhanson/Rectangle#832] [Q&A] [answered]
Hello Community!

Does rectangle support this use-case?

1. Arrange multiple windows across mutliple monitors (desktops)
2. Disconnect and then reconnect the monitors.
3. The windows are now scattered across different monitors.

Is there some way to retain/save the layouts so that when we connect these monitors again, we go back to the same state?

I guess this might not be a trivial feature, but was wondering if such a discussion came up before, and if there is a solution to this.

Thanks!

**Comment by rxhanson** [comment, marked as answer]
Thanks for the feedback.

[Rectangle Pro](https://rectangleapp.com/pro) (paid) is where I've built my solution for this type of scenario. To use it, you'd have to configure an Application group in the `Applications` tab, and you can configure a group to automatically be executed when a display is connected. Note that it is on my todo list to also add a checkbox for when a display is disconnected. You can add window positions for different displays in one application group.

With that said, I'm not opposed to merging a PR in Rectangle for a simpler approach to solve some of these issues if anyone feels inclined to create it.
